packages/ssx-sdk/src/modules/Storage.ts

```diff
@@ -1,42 +1,278 @@
+    const db = await this.getDB();
+    const tx = db.transaction(this.storeName, 'readwrite');
+    const store = tx.objectStore(this.storeName);
+    await store.clear();
+    await tx.done;
   }
 }
-class KeplerDataVault implements IDataVault {
-  private userAuth: IUserAuthorization;
-  private encryption: IEncryption;
 
-  constructor(
-    config: any,
-    userAuth: IUserAuthorization,
-    encryption: IEncryption
-  ) {
-    this.userAuth = userAuth;
+/**
+ * Represents a class for managing browser storage operations with encryption.
+ */
+class BrowserDataVault extends BrowserStorage implements IDataVault {
+  private encryption: IEncryption;
+  /**
+   * Constructs a new instance of BrowserDataVault.
+   * @param config - An object containing optional configuration properties for browser storage.
+   * @param encryption - An instance of an object that implements IEncryption for encryption and decryption operations.
+   */
+  constructor(config: any, encryption: IEncryption) {
+    super(config);
     this.encryption = encryption;
   }
+
+  /**
+   * Retrieves the encrypted stored value for the specified key,
+   * decrypts it, and returns the decrypted value.
+   * @param key - The unique identifier for the stored value.
+   * @returns A Promise that resolves to the decrypted value associated
+   * with the given key or null if the key does not exist.
+   */
+  public async get(key: string): Promise<any> {
+    const encryptedData = await super.get(key);
+    if (encryptedData) {
+      return this.encryption.decrypt(encryptedData);
+    }
+    return null;
+  }
+
+  /**
+   * Stores a value with the specified key, encrypting it before storage.
+   * @param key - The unique identifier for the stored value.
+   * @param value - The value to store under the given key with encryption.
+   * @returns A Promise that resolves when the operation is complete.
+   */
+  public async put(key: string, value: any): Promise<void> {
+    const encryptedData = await this.encryption.encrypt(value);
+    return super.put(key, encryptedData);
+  }
+
+  /**
+   * Retrieves the unencrypted stored value associated with the specified key.
+   * @param key - The unique identifier for the stored value.
+   * @returns A Promise that resolves to the unencrypted value associated with the given key or undefined if the key does not exist.
+   */
+  public unencrypted_get(key: string): Promise<any> {
+    return super.get(key);
+  }
+
+  /**
+   * Stores a value without encryption with the specified key.
+   * @param key - The unique identifier for the stored value.
+   * @param value - The value to store under the given key without encryption.
+   * @returns A Promise that resolves when the operation is complete.
+   */
+  public unencrypted_put(key: string, value: any): Promise<void> {
+    return super.put(key, value);
+  }
 }
 
-export { IStorage, IDataVault, BrowserDataVault, KeplerDataVault };
+// class KeplerDataVault implements IDataVault {
```

Comment:
Do we need all this commented code? If this is only an example TSDocs has a tag `@example` where you can add it.